Code/Engine/GameEngine/StateMachine/StateMachineComponent.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <Core/Messages/EventMessage.h>
+#include <GameEngine/StateMachine/StateMachineResource.h>
+
+struct EZ_GAMEENGINE_DLL ezMsgStateMachineStateChanged : public ezEventMessage
+{
+  EZ_DECLARE_MESSAGE_TYPE(ezMsgStateMachineStateChanged, ezEventMessage);
+
+  ezHashedString m_sNewStateName;
+
+private:
+  const char* GetNewStateName() const { return m_sNewStateName; }
+  void SetNewStateName(const char* szName) { m_sNewStateName.Assign(szName); }
+};
+
+//////////////////////////////////////////////////////////////////////////
+
+class ezStateMachineState_SendMsg : public ezStateMachineState
```

Comment:
I guess this file could use a couple of comments. For example what message this class sends.